README.md

```diff
@@ -0,0 +1,238 @@
+            super(TestMyFacilityHandler, self).setUp()
+    
+        def test_good_file(self):
+            # we expect this to succeed, so if the handler experiences an error, it is considered a
+            # "failed test"
+            handler = self.run_handler(GOOD_NC)
+            pass
+    
+        def test_good_file_with_compliance_check(self):
+            # we also expect this to succeed, since the test file is known be CF compliant
+            handler = self.run_handler(GOOD_NC, check_params={'checks': ['cf']})
+            pass
+    
+        def test_bad_file(self):
+            # we expect this to fail with a 'ComplianceCheckFailedError' exception, since it's not actually
+            # a NetCDF file since we expect this to be a failure, we use run_handler_with_exception to
+            # invert the expected result, so that it treats a success as an undesired outcome, and therefore
+            # a "failed test"
+            handler = self.run_handler_with_exception(ComplianceCheckFailedError, NOT_A_NETCDF)
+            pass
+
+    ```
+
+1. You can now test your handler by simply running the unit tests. There are several ways to run them in the IDE, but you can get fine-grained control over which tests are run by opening the test module, and right-clicking on the test class and clicking _'Run unittests for Unittests for test_handlers.TestMyFacilityHandler'_, or even the individual test methods. This enables the handler class itself to be largely configured
+and tested before leaving the IDE, and proceeding on to integration testing, to run the handler in a "deployed" context.
+
+### Integration testing
+It is perfectly possible to perform this setup and IDE testing without the use of the PO box and chef repo, however the key reason to do this is to shortcut the integration testing and leverage the capabilities of Vagrant/Virtualbox shared folders to source the Python libraries directly from the **same directory** that you are editing in the IDE.
+
+This allows for a much more rapid turnaround in getting your "work in progress" code to actually run in a live development environment (in this case, the PO box).
+
+Note: the following assumes that you are running the PO box from the same **${CHEF_REPO}** as referred to above in the IDE setup
+
+1. Add a watch configuration to the **imos_po_watches** databag in Chef. A watch configuration defines:
+
+    1. **path**: types: JSON=array, Python=list : list of incoming directory paths to be watched and having incoming files routed to this pipeline
+    1. **handler**: types JSON=string, Python=str : name of the handler class to use (the handler class is resolved by looking for this string in the **pipeline.handlers** entry point group and retrieving the corresponding handler object)
+    1. **params**:  types JSON=object, Python=dict : passed directly to the handler class **_\_\_init\_\__** method as **\*\*kwargs** in order to configure the handler parameters
```

Comment:
Maybe add a note that the list of available arguments and their meanings can be seen in the [HandlerBase code](https://github.com/aodn/python-aodncore/blob/master/aodncore/pipeline/handlerbase.py#L148)
It might be worth including a few more of the commonly used ones in the example below (e.g. "notify_params", "include_regexes", "dest_path_function")

Reply:
Definitely needs that documentation, but like you mentioned I was going to put that in aodncore documentation and link to it. Just so that if changed, it doesn't need PRs in multiple repos. 

I think some examples would be good though, so will add them. I can't see include_regexes of dest_path_function ever changing, so should be fine to mention them

Reply:
:+1: I wasn't suggesting duplicating the aodncore documentation here, just linking to it.